### PR TITLE
da_revocation: sample revoked DACs and PAI certs for VID:0xFFF1 PID:0x8001

### PR DESCRIPTION
**Pull request by shubhamdp**
Related to #26582.

Generated the certificates with the help of `chip-cert` (please check the commands used below).
Generated and updated CRL using `openssl ca` functionality.

Next update time is set to 99 years.

<details>
<summary> chip-cert commands used for generating DACs </summary>

```
cd credentials/development/attestation

chip-cert gen-att-cert -t d -c "Matter Dev DAC Revoked 01" -V 0xFFF1 -P 0x8001 --lifetime 4294967295 \
	-C Matter-Development-PAI-FFF1-noPID-Cert.pem -K Matter-Development-PAI-FFF1-noPID-Key.pem \
	-o Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.pem -O Matter-Development-DAC-FFF1-8001-Revoked-01-Key.pem

chip-cert gen-att-cert -t d -c "Matter Dev DAC Revoked 02" -V 0xFFF1 -P 0x8001 --lifetime 4294967295 \
	-C Matter-Development-PAI-FFF1-noPID-Cert.pem -K Matter-Development-PAI-FFF1-noPID-Key.pem \
	-o Matter-Development-DAC-FFF1-8001-Revoked-02-Cert.pem -O Matter-Development-DAC-FFF1-8001-Revoked-02-Key.pem

chip-cert gen-att-cert -t d -c "Matter Dev DAC Revoked 03" -V 0xFFF1 -P 0x8001 --lifetime 4294967295 \
	-C Matter-Development-PAI-FFF1-noPID-Cert.pem -K Matter-Development-PAI-FFF1-noPID-Key.pem \
	-o Matter-Development-DAC-FFF1-8001-Revoked-03-Cert.pem -O Matter-Development-DAC-FFF1-8001-Revoked-03-Key.pem
```
</details>

<details>
<summary> chip-cert commands used for generating PAI and DAC </summary>

```
cd credentials/test/attestation

chip-cert gen-att-cert -t i -c "Matter Test PAI 0xFFF1 no PID Revoked" -V 0xFFF1 --lifetime 4294967295 \
	-C Chip-Test-PAA-FFF1-Cert.pem -K Chip-Test-PAA-FFF1-Key.pem \
	-o Chip-Test-PAI-FFF1-noPID-Revoked-Cert.pem -O Chip-Test-PAI-FFF1-noPID-Revoked-Key.pem

chip-cert gen-att-cert -t d -c "Matter Test DAC" -V 0xFFF1 -P 0x8001 --lifetime 4294967295 \
	-C Chip-Test-PAI-FFF1-noPID-Revoked-Cert.pem -K Chip-Test-PAI-FFF1-noPID-Revoked-Key.pem \
	-o Chip-Test-DAC-FFF1-8001-Signed-By-Revoked-PAI-Cert.pem -O Chip-Test-DAC-FFF1-8001-Signed-By-Revoked-PAI-Key.pem
```
</details>

#### Testing
This PR is the building block for automating the DA revocation implementation.
It adds the test data for generating the device attestation revocation set.

Newly added test data is manually tested by dumping the certs/kets/crl using `openssl [x509 | ec | crl]` command. Also verified if both `.der` and `.pem` pairs outputs the same data.

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

File: credentials/development/attestation/Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.pem
Comment (tcarmelveilleux) [inline credentials/development/attestation/Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.pem:1]: Please wait for a review by @bh3000 befire merging.

**Review by bh3000 — COMMENTED** [review]
Could you add a set of CRLs and certs for the indirect CRL signing use case as well?   

For complete test/example coverage there needs to be versions that includes the CRL entry extension Certificate Issuer as well. Note that there are two approaches here that should be included, the case where each entry has the Certificate Issuer, and the case where the entry's are grouped by Certificate Issuer (multiple entries with the same Certificate Issuer where only the first in the group contain the extension). See section 6.2.4.1. Step 10.a. and RFC 5280 section 5.3.3 for detail. 

Can you add those?

File: credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem
Comment (tcarmelveilleux) [reply credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem:1]: I think the indirect CRLs should be a separate PR.

Comment (tcarmelveilleux) [inline credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem:1]: The CRLs have to be regenerated with the Authority Key Identifier extension matching the CRLSignerCertificates Subject Key Identifier.

Comment (bh3000) [reply credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem:1]: To add this you can update the ca.conf you use to include: 

[ crl_ext ]
authorityKeyIdentifier=keyid:always

and within the [ ca ], or where the default_ca points to add: 
crl_extensions = crl_ext

**Review by andy31415 — COMMENTED** [review]
@shubhamdp please add a `Testing` entry in the summary of this PR and explain how this change was tested.

Since I see no unit test changes, I assume this will be "manually tested via..." and in that case we would like to intentionally set the bar higher on manual tests (to avoid all PRs setting manual because it is easier) so please spend time to add details on how this was tested as well as a justification why it is impossible for this PR to be tested in an automated fashion.

**Comment by shubhamdp** [comment]
> This can be done in a separate PR. No need to do here.

Added a comment on the original issue to keep track of it https://github.com/project-chip/connectedhomeip/issues/26582#issuecomment-2545297462

File: credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem
Comment (shubhamdp) [reply credentials/development/attestation/Matter-Development-PAI-FFF1-noPID-CRL.pem:1]: Thanks @bh3000 for the helping out with it. Added the AKID extension to the CRL.

**Comment by shubhamdp** [comment]
> @shubhamdp please add a `Testing` entry in the summary of this PR and explain how this change was tested.
> 
> Since I see no unit test changes, I assume this will be "manually tested via..." and in that case we would like to intentionally set the bar higher on manual tests (to avoid all PRs setting manual because it is easier) so please spend time to add details on how this was tested as well as a justification why it is impossible for this PR to be tested in an automated fashion.

@andy31415 Sorry, I wasn't aware about the decision in the TT call. I have updated the Testing section in the PR description.

File: credentials/development/attestation/Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.der
Comment (bzbarsky-apple) [inline credentials/development/attestation/Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.der:1]: I thought the contents of this dir were auto-generated by running src/tools/chip-cert/dacs.py.  Is that not the case?  Or if it is, will running that script not affect the new files added here?

Comment (shubhamdp) [reply credentials/development/attestation/Matter-Development-DAC-FFF1-8001-Revoked-01-Cert.der:1]: Moved the certs, keys, and crls to `credentials/test/revoked-attestation-certificates`!

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**
